core/common/state-machine/src/main/java/org/eclipse/edc/statemachine/StateProcessorImpl.java

```diff
@@ -29,8 +29,10 @@
  * to apply the wait strategy or not.
  *
  * @param <T> the entity that is processed
+ * @deprecated please use {@link ProcessorImpl}
  */
-public class StateProcessorImpl<T> implements StateProcessor {
+@Deprecated(since = "0.1.3")
+public class StateProcessorImpl<T> implements Processor {
```

Comment:
At least tractus-x is using this for the EDR state machine, I think this deprecation work helps a lot migrating from version to version (because the `@deprecated` tag in the javadoc text)

Reply:
It seems this begs a broader discussion about SPIs, what we should and should not deprecate etc.